### PR TITLE
added log code for counting digits

### PR DESCRIPTION
**Pull request by futoid**
#### Checklist

- [x] Added description of change
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 
Added constant time code to count the number of digits

### CODE REVIEW

**Comment by futoid** [comment]
Hey @Panquesito7 Can you verify and tell if any changes required?

File: math/finding_number_of_digits_in_a_number.cpp
Comment (Panquesito7) [inline math/finding_number_of_digits_in_a_number.cpp:47]: What's the difference between this function and the other function?

File: math/finding_number_of_digits_in_a_number.cpp
Comment (krishnacx) [inline math/finding_number_of_digits_in_a_number.cpp:48]: TC: O(1)

File: math/finding_number_of_digits_in_a_number.cpp
Comment (krishnacx) [reply math/finding_number_of_digits_in_a_number.cpp:47]: The second implementation function is using library function floor.

File: math/finding_number_of_digits_in_a_number.cpp
Comment (krishnacx) [inline math/finding_number_of_digits_in_a_number.cpp:57]: Empty line can be removed

File: math/finding_number_of_digits_in_a_number.cpp
Comment (futoid) [reply math/finding_number_of_digits_in_a_number.cpp:48]: @krishnacx I have removed unnecessary space. Kindly tell if anything require changes.

File: math/finding_number_of_digits_in_a_number.cpp
Comment (futoid) [reply math/finding_number_of_digits_in_a_number.cpp:47]: @Panquesito7 TCs are different

**Review by tjgurwara99 — DISMISSED** [review]

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please fix the [`clang-tidy`](https://github.com/TheAlgorithms/C-Plus-Plus/actions/runs/3914525867/jobs/6691688974) warnings.
Let us know if you need any help with that. Thanks. 🙂

**Review by futoid — COMMENTED** [review]
decoupled test functions

File: math/finding_number_of_digits_in_a_number.cpp
Comment (futoid) [reply math/finding_number_of_digits_in_a_number.cpp:88]: Done. Let me know if anything else required changes : )

File: math/finding_number_of_digits_in_a_number.cpp
Comment (Panquesito7) [inline math/finding_number_of_digits_in_a_number.cpp:12]:
```suggestion
 * numbers (not integers). The code for that is also included (count_digits).
```

File: math/finding_number_of_digits_in_a_number.cpp
Comment (Panquesito7) [inline math/finding_number_of_digits_in_a_number.cpp:48]:
```suggestion
 * TC: O(1)
```

File: math/finding_number_of_digits_in_a_number.cpp
Comment (Panquesito7) [inline math/finding_number_of_digits_in_a_number.cpp:81]:
```suggestion
/**
 * @brief Self-tests using the `count_digits` function
 * @returns void
 */
static void test_count_digits() {
```

**Comment by futoid** [comment]
> 
[cland-tidy](https://github.com/TheAlgorithms/C-Plus-Plus/actions/runs/3914525867/jobs/6691688974) for this
Possible solutions that I found: _making narrowing conversion intentionally._

Kindly assist with this @Panquesito7

**Comment by Panquesito7** [comment]
Let's discuss this on the [Discord community](https://the-algorithms.com/discord), so that other members can help as well. 🙂

File: math/finding_number_of_digits_in_a_number.cpp
Comment (tjgurwara99) [inline math/finding_number_of_digits_in_a_number.cpp:64]:
```suggestion
    return floor(log10(n) + 1); // NOLINT(cppcoreguidelines-narrowing-conversions)
```

Though I would say this is not a good practice, and I would rather you make the function return type `count_digits` to be `double` instead. But I leave that up to @Panquesito7 to decide 😄

Comment (futoid) [reply math/finding_number_of_digits_in_a_number.cpp]: I've changed the return type to avoid clang-tidy warnings

**Review by Panquesito7 — APPROVED** [review]
Amazing. Thanks! 🚀

**Review by realstealthninja — DISMISSED** [review]
Looks good to me

File: math/finding_number_of_digits_in_a_number.cpp
Comment (tjgurwara99) [inline math/finding_number_of_digits_in_a_number.cpp:62]: Use guard if you already know what to return. Looking at your second `if` statement, it's clear that you already know what to `return` so use that as guard instead of `else if`.

 ```suggestion
    if (n == 0) {
        // handling the value 0
        return 0;
    }
    if (n < 0) {
        n = -n;
    }
```

**Review by tjgurwara99 — CHANGES_REQUESTED** [review]

File: math/finding_number_of_digits_in_a_number.cpp
Comment (tjgurwara99) [reply math/finding_number_of_digits_in_a_number.cpp]: Please try to not @ mention people. It creates a lot of unnecessary noise for maintainers 🙏🏼

**Review by tjgurwara99 — APPROVED** [review]

**Review by Panquesito7 — APPROVED** [review]

**Pull request merged**
